firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/CrashlyticsReport.java

```diff
@@ -98,6 +97,10 @@ public static Builder builder() {
    */
   @NonNull
   public CrashlyticsReport withEvents(@NonNull ImmutableList<Event> events) {
+    if (this.getNdkPayload() != null) {
```

Comment:
Is there any concern that we might have a situation where this is null, but session is as well?

Reply:
I see what you mean, so check for getSession() == null and do illegal state then? sounds good

Reply:
👍